src/graph/graph.c

```diff
@@ -987,60 +1003,63 @@ uint64_t Graph_GetNodeDegree
 
 	// for each relationship type to consider
 	for(edgeType = start_rel; edgeType < end_rel; edgeType++) {
-		M = Graph_GetRelationMatrix(g, edgeType, false);
-
 		//----------------------------------------------------------------------
 		// outgoing edges
 		//----------------------------------------------------------------------
 
-		// TODO: revisit once we get rid of MULTI-EDGE hack
+		M = Graph_GetRelationMatrix(g, edgeType, false);
+		ME  = Graph_MultiEdgeRelationMatrix(g, edgeType);
+		
 		if(outgoing) {
 			// construct an iterator to traverse over the source node row,
 			// containing all outgoing edges
-			RG_MatrixTupleIter_AttachRange(&it, M, srcID, srcID);
+			Delta_MatrixTupleIter_AttachRange(&it, M, srcID, srcID);
 			// scan row
-			while(RG_MatrixTupleIter_next_UINT64(&it, NULL, &destID, &edgeID)
+			while(Delta_MatrixTupleIter_next_UINT64(&it, NULL, NULL, &edgeID)
 					== GrB_SUCCESS) {
-
 				// check for edge type single/multi
 				if(SINGLE_EDGE(edgeID)) {
 					edge_count++;
 				} else {
-					// multiple edges connecting src to dest
-					// entry is a pointer to an array of edge IDs
-					EdgeID *multi_edge = (EdgeID *)(CLEAR_MSB(edgeID));
-					edge_count += array_len(multi_edge);
+					EdgeID me_id = CLEAR_MSB(edgeID);
+					Delta_MatrixTupleIter me_it;
+					Delta_MatrixTupleIter_AttachRange(&me_it, ME, me_id, me_id);
+					while(Delta_MatrixTupleIter_next_BOOL(&me_it, NULL, NULL, NULL) == GrB_SUCCESS) {
+						edge_count++;
+					}
+					Delta_MatrixTupleIter_detach(&me_it);
 				}
 			}
-			RG_MatrixTupleIter_detach(&it);
+			Delta_MatrixTupleIter_detach(&it);
 		}
 
 		//----------------------------------------------------------------------
 		// incoming edges
 		//----------------------------------------------------------------------
 
 		if(incoming) {
-			// transposed relation matrix
 			TM = Graph_GetRelationMatrix(g, edgeType, true);
-
 			// construct an iterator to traverse over the source node row,
 			// containing all incoming edges
-			RG_MatrixTupleIter_AttachRange(&it, TM, srcID, srcID);
-			while(RG_MatrixTupleIter_next_BOOL(&it, NULL, &destID, NULL)
+			Delta_MatrixTupleIter_AttachRange(&it, TM, srcID, srcID);
+			// scan row
+			while(Delta_MatrixTupleIter_next_BOOL(&it, NULL, &destID, NULL)
 					== GrB_SUCCESS) {
-
+				Delta_Matrix_extractElement_UINT64(&edgeID, M, destID, srcID);
```

Comment:
Please add a remark saying that for every incoming edge we need to perform a binary search to locate the actual edge ID.